### DISCUSSION TITLE
How to access data?

### DISCUSSION THREAD

**Discussion by StijnCoolen** [nuxt/nuxt#17003] [Questions] [answered]
For some reason I cannot access anything inside the `data` inside my script. I cannot use the `ref `object and when I try to get it's value, it returns a `Proxy`.
```

<script setup>
  const { data } = await useAsyncData('articleData', () => $api.getArticleDataBySlug(route.params.slug));
  console.log(data); -> ref object
  console.log(data.value); -> Proxy
</script>
```

What am I doing wrong? (I do see the data being logged in the terminal/server, but not client-side)

**Comment by TheAlexLichter** [comment]


**Reply by StijnCoolen** [reply]
The return value is a `Promise`. The function itself looks like this: 
```
getArticleDataBySlug: async(slug) => {
                    try {
                        const response = await fetchGQL(articleQuery, {
                            variables: {
                                slug
                            }
                        });

                        return response;
                    } catch (error) {
                        console.error('Error fetching article by slug:', error);
                        return { error };
                    }
                },
```

**Comment by crotanite** [comment, marked as answer]
Sometimes when I `console.log` data I also get a Proxy, but if you look under [[Target]] you should see the data and should be able to get it through `data.value.whatever`. If not, can you screenshot/show what you're seeing instead?

**Reply by StijnCoolen** [reply]
I think this is indeed the case.
